Problem1.java

```diff
@@ -0,0 +1,81 @@
+// Time Complexity :O(log(n))
+// Space Complexity :O(1)
+// Did this code successfully run on Leetcode : yes
```

Comment:
If you observe, the logic for firstOccurence and lastOccurence are same. Use a flag to avoid code duplication